weave/trace/weave_client.py

```diff
@@ -1114,37 +1118,35 @@ def send_score_call() -> str:
             call_ref = get_ref(predict_call)
             if call_ref is None:
                 raise ValueError("Predict call must have a ref")
-            call_ref_uri = call_ref.uri()
+            weave_ref_uri = call_ref.uri()
             scorer_call_ref = get_ref(score_call)
             if scorer_call_ref is None:
                 raise ValueError("Score call must have a ref")
             scorer_call_ref_uri = scorer_call_ref.uri()
-            scorer_op_ref_uri = score_call.op_name
-            scorer_op_ref = parse_uri(scorer_op_ref_uri)
-            if not isinstance(scorer_op_ref, OpRef):
-                raise TypeError(f"Invalid scorer op ref: {scorer_op_ref_uri}")
-            score_name = scorer_op_ref.name
+
+            # If scorer_object_ref_uri is provided, it is used as the runnable_ref_uri
+            # Otherwise, we use the op_name from the score_call. This should happen
+            # when there is a Scorer subclass that is the source of the score call.
+            runnable_ref_uri = scorer_object_ref_uri or score_call.op_name
             score_results = score_call.output
 
-            return self._add_score(
-                call_ref_uri=call_ref_uri,
-                score_name=score_name,
-                score_results=score_results,
-                scorer_call_ref_uri=scorer_call_ref_uri,
-                scorer_op_ref_uri=scorer_op_ref_uri,
+            return self._add_runnable_feedback(
+                weave_ref_uri=weave_ref_uri,
+                output=score_results,
+                call_ref_uri=scorer_call_ref_uri,
+                runnable_ref_uri=runnable_ref_uri,
             )
 
         return self.future_executor.defer(send_score_call)
 
     @trace_sentry.global_trace_sentry.watch()
-    def _add_score(
+    def _add_runnable_feedback(
         self,
         *,
+        weave_ref_uri: str,
```

Comment:
we are starting to different URIs in a lot of places.  Maybe worth type narrowing to avoid errors

Reply:
what do you mean?

Reply:
Is there a difference in shape between `weave_ref_uri` and `call_ref_uri`?  If we have many different types of URI floating around, can we help ourselves use the correct type of URI?

Reply:
Hmmm, let's make sure we are on the same page:
  * The Feedback table now has a few key column (they all have the `_ref` suffix and are strings formatted like refs:
    * `weave_ref`
    * `call_ref`
    * `trigger_ref`
    * `runnable_ref`
    * `annotation_ref`
  * At runtime, I like to distinguish between runtime Refs (instances of Refs) and the string uri by using `_ref_uri` to make it super clear this is a string, not the runtime object.

Given that context, i am having a hard time groking your recommendation or callout. Can you elaborate?

Reply:
Maybe I'm saying something silly or overkill

Here's what I observe:
1. We are using URIs in a lot of places in the SDK
2. They can have different shapes like [(1)](https://github.com/wandb/weave/blame/tim/basic_batch_actions/weave/trace/refs.py#L75), [(2)](https://github.com/wandb/weave/blame/tim/basic_batch_actions/weave/trace/refs.py#L148), [(3)](https://github.com/wandb/weave/blame/tim/basic_batch_actions/weave/trace/refs.py#L210), and a specific shape of URI maps to a specific type of Ref in our SDK.
3. Should we make a subtype of str for the URIs to avoid passing the wrong URI?

For example:
```py
T = TypeVar('T', bound='WeaveURI')

class WeaveCallURI(str):
    def __new__(cls: Type[T], content: str) -> T:
        # validate it conforms to f"weave:///{self.entity}/{self.project}/call/{self.id}"
       return cast(T, super().__new__(cls, content))
```